tests/test_denest.py

```diff
@@ -184,3 +184,87 @@ def test__schema__nested_arrays_add_tables():
+                  "properties": {
+                      "b": {
+                          "type": "array",
+                          "items": {
+                              "type": "object",
+                              "properties": {
+                                  "c": {
+                                      "type": "object",
+                                      "properties": {
+                                          "d": {"type": "integer"},
+                                          "e": {"type": "array",
+                                                "items": {"type": "object",
+                                                          "properties": {
+                                                              "f": {"type": "string"},
+                                                              "g": {"type": "boolean"}}}}}}}}}}}}},
+        [],
+        [{"a": {"b": []}},
+         {"a": {"b": [{"c": {"d": 1}}]}},
+         {"a": {"b": [{"c": {"d": 12}},
+                      {"c": {"d": 123}}]}},
+         {"a": {"b": [{"c": {"d": 1234}},
+                      {"c": {"d": 12345}},
+                      {"c": {"d": 123456}}]}},
+         {"a": {"b": [{"c": {"e": [{"f": "hello",
+                                    "g": True},
+                                   {"f": "goodbye",
+                                    "g": True}]}}]}}])
+
+    print('denested:', denested)
+
+    assert 3 == len(denested)
+
+    root_table_checked = False
+    nested_table_checked = False
+    nested_nested_table_checked = False
+
+    for table_batch in denested:
+        if tuple() == table_batch['streamed_schema']['path']:
+            root_table_checked = True
+
+            assert {} == table_batch['streamed_schema']['schema']['properties']
+
+            assert 5 == len(table_batch['records'])
+
+            for record in table_batch['records']:
+                assert {} == record
+
+    for table_batch in denested:
+        if ('a', 'b') == table_batch['streamed_schema']['path']:
+            nested_table_checked = True
+
+            assert {'type': 'integer'} == table_batch['streamed_schema']['schema']['properties'][('c', 'd')]
+
+            assert 7 == len(table_batch['records'])
+
+            for record in table_batch['records']:
+                assert 'integer' == record[('c', 'd')][0]
+                assert int == type(record[('c', 'd')][1])
+
+    for table_batch in denested:
+        if ('a', 'b', 'c', 'e') == table_batch['streamed_schema']['path']:
+            nested_nested_table_checked = True
+
+            assert {'type': 'string'} == table_batch['streamed_schema']['schema']['properties'][('f',)]
+            assert {'type': 'boolean'} == table_batch['streamed_schema']['schema']['properties'][('g',)]
+
+            assert 2 == len(table_batch['records'])
+
+            for record in table_batch['records']:
+                assert 'string' == record[('f',)][1]
+                assert record[('g',)][1]
```

Comment:
I would change this to:
```python
for record in table_batch['records']:
    assert 'string' == record[('f',)][0]
    assert str == type(record[('f',)][1])

    assert 'boolean' == record[('g',)][0]
    assert bool == type(record[('g',)][1])
```

Reply:
Just to have it consistent with what we have in `nested_table_checked` loop:
```python
assert 'integer' == record[('c', 'd')][0]
assert int == type(record[('c', 'd')][1])
```

Reply:
Good call